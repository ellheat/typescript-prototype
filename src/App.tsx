import React from 'react';
import TextField from '@material-ui/core/TextField';
import Pagination from '@material-ui/lab/Pagination';
import './App.css';

import { RepositoriesTable } from './components/repositoriesTable';

import { useSearch } from './hooks/useSearch';
import { FETCH_STATUS, useRepositories } from './hooks/useRepositories';
import { usePagination } from './hooks/usePagination';


function App() {
  const [{ searchValue }, { handleChangeSearchValue }] = useSearch();
  const [{ page }, { handleChangePage }] = usePagination(searchValue);
  const { status, pagesCount, repositories } = useRepositories(searchValue, page);

  return (
    <div className="App">
      <TextField id="searchField" label="Search" name="searchField" onChange={handleChangeSearchValue} value={searchValue} />
      { status === FETCH_STATUS.loading && 'Loading...' }
      { status === FETCH_STATUS.success && (
        <>
          <RepositoriesTable repositories={repositories} />
          <Pagination count={pagesCount} color="primary" page={page} onChange={handleChangePage}  />
        </>
      ) }
      { status === FETCH_STATUS.error && 'Something went wrong' }
    </div>
  );
}

export default App;
