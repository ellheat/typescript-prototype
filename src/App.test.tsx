import React from 'react';
import { render, screen } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import App from './App';

describe('App: Component', () => {
  const component = () => (
    <QueryClientProvider client={new QueryClient()}>
      <App />
    </QueryClientProvider>
  );

  test('should render correctly', () => {
    const { container } = render(component());
    expect(container).toMatchSnapshot();
  });

  test('should find page title equals Search Repo from GitHub', () => {
    render(component());
    const titleElement = screen.getByText(/Search Repo from GitHub/i);
    expect(titleElement).toBeInTheDocument();
  });
});
