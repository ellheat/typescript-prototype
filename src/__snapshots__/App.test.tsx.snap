// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: Component should render correctly 1`] = `
<div>
  <div
    class="sc-bdnxRM iiGFbS"
  >
    <h1
      class="sc-gtsrHT"
    >
      Search Repo from GitHub
    </h1>
    <div
      class="MuiFormControl-root MuiTextField-root sc-dlnjwi DTPRZ"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
        for="searchField"
        id="searchField-label"
      >
        Search
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          id="searchField"
          name="searchField"
          type="text"
          value=""
        />
      </div>
    </div>
    <p
      class="sc-eCApnc cDHgXe"
    >
      Type to find repo
    </p>
  </div>
</div>
`;
